src/networkfilemanager.cpp

```diff
@@ -1922,6 +1939,267 @@ static const char *pj_curl_get_header_value(PJ_CONTEXT *,
+    auto fetching = [&]() {
+        emscripten_fetch_t *fetch = nullptr;
+        while (true) {
+
+            emscripten_fetch_attr_t attr;
+            emscripten_fetch_attr_init(&attr);
+            strcpy(attr.requestMethod, "GET");
+            const char *requestHeaders[] = {"Range", szBuffer, nullptr};
+            attr.requestHeaders = requestHeaders;
+            attr.attributes = EMSCRIPTEN_FETCH_LOAD_TO_MEMORY |
+                              EMSCRIPTEN_FETCH_SYNCHRONOUS |
+                              EMSCRIPTEN_FETCH_REPLACE;
+            if (fetch) {
+                emscripten_fetch_close(fetch);
+            }
+
+            TRACE_FETCH("Pre fecth. url: " << url << " range: " << szBuffer);
+            fetch = emscripten_fetch(&attr, url.c_str());
+            // emscripten_fetch_wait(fetch, -1); // This is deprecated
+            TRACE_FETCH("Post fecth");
+
+            if (!fetch) {
+                snprintf(out_error_string, error_string_max_size,
+                         "Cannot init emscripten_fetch for url %s",
+                         url.c_str());
+                return;
+            }
+            const auto response_code = fetch->status;
+            TRACE_FETCH("response code: " << response_code);
+            if (response_code == 0 || response_code >= 300) {
+                const double delay =
+                    GetNewRetryDelay(static_cast<int>(response_code), oldDelay,
+                                     fetch->data, fetch->statusText);
+                if (delay != 0 && delay < MAX_RETRY_DELAY_MS) {
+                    pj_log(ctx, PJ_LOG_TRACE,
+                           "Got a HTTP %ld error. Retrying in %d ms",
+                           response_code, static_cast<int>(delay));
+                    TRACE_FETCH("Got a HTTP " << response_code
+                                              << " retry in ms " << delay);
+
+                    sleep_ms(static_cast<int>(delay));
+                    oldDelay = delay;
+                } else {
+                    if (out_error_string) {
+                        if (fetch->statusText[0]) {
+                            snprintf(out_error_string, error_string_max_size,
+                                     "%s", fetch->statusText);
+                        } else {
+                            snprintf(out_error_string, error_string_max_size,
+                                     "HTTP error %hu: %s", response_code,
+                                     fetch->data);
+                        }
+                    }
+                    TRACE_FETCH("HTTP  error for " << url);
+                    emscripten_fetch_close(fetch);
+                    return;
+                }
+            } else {
+                break;
+            }
+        } // end of while(true)
+
+        if (out_error_string && error_string_max_size) {
+            out_error_string[0] = '\0';
+        }
+
+        // cast 64b to 32b. Hopefully we dont download more than 2GB
```

Comment:
is that comment accurate ? I believe 64-bit WASM builds are possible

Reply:
Oh, pretty recent: https://webassembly.org/news/2025-09-17-wasm-3.0/